lib/google/gapic/generator/util.rb

```diff
@@ -0,0 +1,111 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'protobuf/descriptors'
+
+module Google
+  module Gapic
+    module Generator
+
+      # The utility class that is passed as context to the templates.
+      #
+      # This class also exposes utility templates provided by the template
+      # provider. These utility templates are exposed as Erubis::Eruby objects
+      # which can be accessed by the the name of the symbol.
+      class Util
+        # This token signifies that the a new file will is being rendered. The
+        # content following this token is understood to be the content of the
+        # file.
+        NAME_TOKEN = '##FILE##'.freeze
+
+        # Initializes the Util.
+        def initialize template_provider
+          load_templates template_provider
+        end
+
+        # Exposes the templates.
+        def method_missing(name, *args, &block)
+          return @templates[name] if @templates[name]
+          super
+        end
+
+        # Declares the start of a new file.
+        #
+        # @param name [String] The name of the new file.
+        # @returns [String] The declaration of a new file.
+        def file_name name
+          raise ArgumentError.new 'A name must be passed to #file_name' unless name
+          "#{NAME_TOKEN}#{name}"
+        end
+
+        # Converts a camelCased string to a snake_cased string.
+        def snake_case s
```

Comment:
Check if there is a library for this string conversion stuff.

Reply:
Hmm. Do we really want to bring in ActiveSupport just for this?

Reply:
For something like a generator I would have no problem depending on it.

Reply:
Eh, okay, not a big deal. (I own that I have a terrible personal bias against ActiveSupport.)

Reply:
https://github.com/googleapis/gapic-generator-ruby/issues/10